src/main/java/perosnal/spotifymatcher/service/UserService.java

```diff
@@ -90,15 +96,14 @@ public List<User> getMatches(String accessToken) {
                 .toList();
     }
 
-    public List<Track> getTracksDetails(String accessToken) {
-        User user = userRepository.getById(spotifyApiService.getIdByToken(accessToken));
-        return spotifyApiService.getTracksDetails(user.getTracks(), accessToken);
-    }
-
-    public List<String> getTopTracks(String accessToken) {
-        return userRepository.getById(spotifyApiService.getIdByToken(accessToken))
-                .getTracks();
+    public List<Track> getTracksDetails(String jwt) {
+        jwtTokenValidator.validateJwt(jwt);
+        User user = userRepository.getById(jwtTokenValidator.getUserSpotifyId(jwt));
+        List<Track> tracks = new ArrayList<>();
+        for (String id : user.getTracks()) {
+            tracks.add(trackRepository.getById(id));
```

Comment:
Tracks from repository should be fetched with a batch query, and not individually.